### DISCUSSION TITLE
How about adding a page length feature on table ?

### DISCUSSION THREAD

**Discussion by t-prod** [livewire/flux#490] [Feature Requests]
Hi,

Frequently on datatables we have a page length feature to select how many rows we should have to display on each page ?
Is there something that we can do in Flux ?
We must publish the pagination component to do so but also have the paginate variable dynamic ?

Someone has an idea to how ti implement this ?

Cyril

**Comment by jeffchown** [comment]
@t-prod I simply do this when returning the records to be displayed in the table:

```
    #[Computed()]
    public function clients(): Collection|LengthAwarePaginator
    {
        return Client::get()->paginate(config('app.item_per_page'));
    }
```

then in my blade, of course is `@foreach ($this->clients as $client)`

**Comment by t-prod** [comment]
Sorry I was not clear ;)
I speak about this kind of feature :
![image](https://github.com/user-attachments/assets/8154fbf4-5e1c-4a03-863a-320ad26975e5)

**Reply by jeffchown** [reply]
Ahhh, I see.  Thanks for the clarification @t-prod 👍

**Comment by t-prod** [comment]
Unfortunately it should be on pagination table but we could create a flux dropdown like this to do so and make the lengths in config app as you do ;)

![image](https://github.com/user-attachments/assets/7c9779ef-a9dc-47bd-8028-fcc6f1c94add)

Thanks again for your help @jeffchown

**Comment by dicktornfeldt** [comment]
I also would like a feature like this. Currently I've solved it like this, but it doesn't look Fluxy :) 

![Screenshot 2025-03-06 at 10 18 36](https://github.com/user-attachments/assets/59177bb5-0ead-46a3-949a-56b190384454)

**Comment by TechDeptLunaSoftwareGroup** [comment]
I would like this to... Right now I am putting it at the top.  But sometimes I can have many filters and adding that to the top just makes it more messy and limits the room I can work with.

**Comment by bobik1337** [comment]
Agree. Need it very much

**Comment by bobik1337** [comment]
Up here. Will try to do it myself. If it would go well, i will share the code

**Comment by bobik1337** [comment]
Finally, I've got some spare time to make it myself.

Here is the guide:
1. Publish (or copy) paginaton.blade.php
2. add to props the code below:
```
    'perPageOptions' => [15, 25, 50, 100],
```
3. Change the code since ~50 row
```
@else
    <div {{ $attributes->class('@container pt-3 border-t border-zinc-100 dark:border-zinc-700 flex justify-between items-center gap-3') }} data-flux-pagination>
        @if ($paginator->total() > 0)
            <div class="flex items-center gap-3">
                <div class="text-zinc-500 dark:text-zinc-400 text-xs font-medium whitespace-nowrap">
                    {{ $paginator->firstItem() }}-{{ $paginator->lastItem() }} {!! __('of') !!} {{ $paginator->total() }}
                </div>
                @if ($paginator->total() > 15)
                    <div class="flex items-center gap-2">
                        <flux:select wire:model.live="perPage" size="sm" class="w-20">
                            @foreach ($perPageOptions as $option)
                                <flux:select.option value="{{ $option }}">{{ $option }}</flux:select.option>
                            @endforeach
                        </flux:select>
                        <div class="text-zinc-500 dark:text-zinc-400 text-xs font-medium whitespace-nowrap">per page</div>
                    </div>
                @endif
            </div>
        @else
            <div></div>
        @endif
```
It would look like this
<img width="351" height="255" alt="image" src="https://github.com/user-attachments/assets/c8f9eb25-52e7-4fa6-907b-bec2c2ccc964" />
4. Now you can use it. Just add to your component something like this:
```
    public int $perPage = 15;

    public function updatedPerPage(): void
    {
        $this->resetPage();
    }

        $expensesPage = $query
            ->orderBy('date', 'desc')
            ->orderBy('id', 'desc')
            ->paginate($this->perPage);
```

Still would be happy if it was flux-native, as it's not yet greatly aligned

**Comment by slovenianGooner** [comment]
I handle it a bit differently without publishing the flux::pagination component.

```blade
<div class="px-3 py-3 flex gap-4 justify-start items-center">
    <div>
        @if ($this->canSelectPerPage ?? false)
            <flux:dropdown>
                @if ($this->perPage === 'all')
                    <flux:button size="xs" icon-trailing="chevron-down">
                        {{ __(UI::ALL) }}
                    </flux:button>
                @else
                    <flux:button size="xs" icon-trailing="chevron-down">
                        {{ __(UI::PER_PAGE) }}: {{ $this->perPage }}
                    </flux:button>
                @endif
                <flux:menu>
                    <flux:menu.radio.group wire:model.live="perPage">
                        @foreach ($this->getPerPageOptions() as $key => $value)
                            <flux:menu.radio :value="$key">{{ $value }}</flux:menu.radio>
                        @endforeach
                    </flux:menu.radio.group>
                </flux:menu>
            </flux:dropdown>
        @endif
    </div>
    @if ($this->rows instanceof LengthAwarePaginator)
        <div class="flex-1 *:data-flux-pagination:space-x-6 *:data-flux-pagination:pt-0 *:data-flux-pagination:border-0 *:data-flux-pagination:justify-end!">
            <flux:pagination :paginator="$this->rows"/>
        </div>
    @endif
</div>
```

The result is as follows

<img width="1232" height="132" alt="Screenshot 2026-02-11 at 21 30 57" src="https://github.com/user-attachments/assets/267c351e-55f3-4900-9b38-fbfe9c807569" />
